os/tools/mkbootparam.py

```diff
@@ -58,105 +64,115 @@
-        print "FAIL!! Bootparam should be located at the end of flash with 8K. Please re-configure."
-        sys.exit(1)
-
-    # Get addresses of kernel partitions
-    kernel_address = []
-    offset = 0
-    for index, types in enumerate(names):
-        if types == "kernel":
-            address = hex(int(FLASH_START_ADDR, 16) + offset)
-            kernel_address.append(address)
-        offset += int(sizes[index]) * 1024
-
-    if len(kernel_address) == 0 or len(kernel_address) > 2:
-        print "FAIL!! No found kernel partition"
-        sys.exit(1)
-
-    if os.path.exists(user_file_dir):
-        user_file_list = os.listdir(user_file_dir)
-    else:
-        user_file_list = []
-    user_app_count = len(user_file_list)
-    SIZE_OF_BINNAME = 16
-    SIZE_OF_BINIDX = 1
-    app_bootparam_size = (user_app_count * (SIZE_OF_BINNAME + SIZE_OF_BINIDX)) + 1
-
-    with open(bootparam_file_path, 'wb') as fp:
-        # Write Data
-        bootparam_version = 1
-        bootparam_format_version = 1
-        kernel_active_idx = 0
-        fp.write(struct.pack('I', bootparam_version))
-        fp.write(struct.pack('I', bootparam_format_version))
-	# Kernel data
-        fp.write(struct.pack('B', kernel_active_idx))
-	for address in kernel_address:
-            fp.write(struct.pack('I', int(address, 16)))
-    # Add checksum
-    mkchecksum_path = os.path.dirname(__file__) + '/mkchecksum.py'
-    os.system('python %s %s' % (mkchecksum_path, bootparam_file_path))
-
-    with open(bootparam_file_path, 'a') as fp:
-        # User Data
-        user_active_idx = 0
-        fp.write(struct.pack('B', user_app_count))
-        for user_file in user_file_list:
-            fp.write('{:{}{}.{}}'.format(user_file, '<', SIZE_OF_BINNAME, SIZE_OF_BINNAME - 1).replace(' ','\0'))
-            fp.write(struct.pack('B', user_active_idx))
-
-    # Fill remaining space with '0xff'
-    with open(bootparam_file_path, 'a') as fp:
-        remain_size = partition_size - bootparam_size - app_bootparam_size
-        fp.write(b'\xff' * remain_size)
+	print "========== Start to make boot parameters =========="
+	SIZE_OF_CHECKSUM = 4
+	SIZE_OF_BP_VERSION = 4
+	SIZE_OF_BP_FORMAT_VERSION = 4
+	SIZE_OF_KERNEL_INDEX = 1
+	SIZE_OF_KERNEL_FIRST_ADDR = 4
+	SIZE_OF_KERNEL_SECOND_ADDR = 4
+
+	kernel_data_size = SIZE_OF_CHECKSUM + SIZE_OF_BP_VERSION + SIZE_OF_BP_VERSION + SIZE_OF_KERNEL_INDEX + SIZE_OF_KERNEL_FIRST_ADDR + SIZE_OF_KERNEL_SECOND_ADDR
+
+	FLASH_START_ADDR = get_config_value(config_file_path, "CONFIG_FLASH_START_ADDR=")
+	FLASH_SIZE = get_config_value(config_file_path, "CONFIG_FLASH_SIZE=")
+	names = get_config_value(config_file_path, "CONFIG_FLASH_PART_NAME=").replace('"','').replace('\n','').split(",")
+	sizes = get_config_value(config_file_path, "CONFIG_FLASH_PART_SIZE=").replace('"','').replace('\n','').split(",")
+	names = filter(None, names)
+	sizes = filter(None, sizes)
+
+	# Calculate partition size of boot parameters
```

Comment:
I think this is a verification of bootparam partition size and its position.
So it is better to make function for this and change the comment about verification.